stdlib/public/Concurrency/DispatchGlobalExecutor.inc

```diff
@@ -234,9 +300,8 @@ static void swift_task_enqueueGlobalWithDeadlineImpl(long long sec,
                                                      long long tnsec,
                                                      int clock, Job *job) {
   assert(job && "no job provided");
-
-  dispatch_function_t dispatchFunction = &__swift_run_job;
-  void *dispatchContext = job;
+  auto task = dyn_cast<AsyncTask>(job);
+  assert(task && "provided job must be a task");
```

Comment:
There's a `cast<>` which asserts.

I'm not sure it's okay to assume we're given a task here, but you could definitely use the task allocator instead of `malloc` when we are.

Reply:
all cases that this function is ever called are from full-on task based initialization and not child tasks or other such jobs. I will change to `cast<>`.